crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -42,19 +49,32 @@ pub use bevy_ecs_macros::States;
 /// }
 ///
 /// ```
-pub trait States: 'static + Send + Sync + Clone + PartialEq + Eq + Hash + Debug {}
+pub trait States: 'static + Send + Sync + Clone + PartialEq + Eq + Hash + Debug {
+    /// How many other states this state depends on.
+    /// Used to help order transitions and de-duplicate [`ComputedStates`], as well as prevent cyclical
+    /// `ComputedState` dependencies.
+    const DEPENDENCY_DEPTH: usize = 1;
```

Comment:
For normal states - nothing, their update will just run later, and any states that depend on them will run later as well.

For sub-states - it is implemented in the macro, but they manually implement it they could end up having their update called at before all their dependencies have been updated - so they will operate on stale information. So - not a panic, but still incorrect. However - since we can't auto-implement it for both SubStates & Computed States, we have to rely on 
a default implementation that users can change...

For computed states - it is auto-implemented, so it requires no additional work on their part.

Reply:
Correction: for normal states, it will have no effect. It'll just push any states that depend on them later in the order of schedule updates., but that doesn't have any functional impacts. (so the conclusion is the same, just the mechanism is different)

Sorry for the mistake there - it's been a while since I looked at this code, so I'm re-learning it as I'm working to find a resolution to your concerns around executors.